pkg/commands/workflows/plan_status_comments.go

```diff
@@ -122,10 +122,20 @@ func (c *PlanStatusCommentCommand) Run(ctx context.Context, args []string) error
 
 // Process handles the main logic for the Guardian plan status comments process.
 func (c *PlanStatusCommentCommand) Process(ctx context.Context) error {
+	failureResultCodes := []github.GitHubWorkflowResult{
+		github.GitHubWorkflowResultCancelled,
+		github.GitHubWorkflowResultFailure,
+	}
+
 	// there was at least one failure, we should return an error to fail the job
 	// no comments as each plan run will comment their failure status
-	if c.flagInitResult == github.GitHubWorkflowResultFailure || slices.Contains(c.flagPlanResult, github.GitHubWorkflowResultFailure) {
-		return fmt.Errorf("init or plan has one or more failures")
+	if slices.Contains(failureResultCodes, github.GitHubWorkflowResult(c.flagInitResult)) {
+		return fmt.Errorf("init was not successful")
+	}
+	if slices.ContainsFunc(c.flagPlanResult, func(result string) bool {
+		return slices.Contains(failureResultCodes, github.GitHubWorkflowResult(result))
+	}) {
+		return fmt.Errorf("plan has one or more failures")
```

Comment:
Perhaps its best to rename appropriately... 
```suggestion
		return fmt.Errorf("plan has one or more failures or was cancelled")
```
We could also check the status codes and return `failures` if a failure occurred or `plan was cancelled` if it was cancelled. That might be better for the user.